sklearn/ensemble/iforest.py

```diff
@@ -210,18 +210,18 @@ def predict(self, X):
         """
         # code structure from ForestClassifier/predict_proba
         # Check data
-        X = self.estimators_[0]._validate_X_predict(X, check_input=True)
+        X = check_array(X, accept_sparse='csr')
         n_samples = X.shape[0]
 
-
         n_samples_leaf = np.zeros((n_samples, self.n_estimators), order="f")
         depths = np.zeros((n_samples, self.n_estimators), order="f")
 
-        for i, tree in enumerate(self.estimators_):
-            leaves_index = tree.apply(X)
-            node_indicator = tree.decision_path(X)
+        for i, (tree, features) in enumerate(zip(self.estimators_,
+                                                 self.estimators_features_)):
+            leaves_index = tree.apply(X[:, features])
```

Comment:
Some operations on arrays will cause a copy of that array, so you end up using more memory than needed. I tried to find a good reference on how to determine if two arrays share data or not, but the conclusion seems to be "it is complicated".